datafusion/datasource-csv/src/file_format.rs

```diff
@@ -560,21 +572,27 @@ impl CsvFormat {
                     })
                     .unzip();
             } else {
-                if fields.len() != column_type_possibilities.len() {
-                    return exec_err!(
-                            "Encountered unequal lengths between records on CSV file whilst inferring schema. \
-                             Expected {} fields, found {} fields at record {}",
-                            column_type_possibilities.len(),
-                            fields.len(),
-                            record_number + 1
-                        );
```

Comment:
Would it be beneficial to keep this error message for when `truncated_rows` config is `false`? What does the error message look like right now if we read CSV files with differing column counts where this new behaviour is not enabled?